test/lit/passes/optimize-instructions-gc.wast

```diff
@@ -386,29 +386,23 @@
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   (block
   ;; CHECK-NEXT:    (drop
-  ;; CHECK-NEXT:     (ref.as_non_null
-  ;; CHECK-NEXT:      (local.get $data)
-  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:     (local.get $data)
```

Comment:
Why is this optimized? It doesn't have instructions like `struct.get`, no?

Reply:
This began as `(ref.as_func (ref.as_non_null (local.get $data)))`, and the `ref.as_func` will trap on null anyhow, so the `ref.as_non_null` in the middle can be removed.

(And after that, we see that the `ref.as` will always fail, so it drops the input and adds an unreachable.).